### PR TITLE
acl: remove ACLDisabledTTL

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #10808 , related to this comment: https://github.com/hashicorp/consul/pull/10808#discussion_r684543725

Details in commit messages.

This PR removes an unused config field and reduces the implementation of the `ACLResolver` around disabling of ACLs.

### CODE REVIEW

File: agent/consul/acl.go [hunk 1]
```diff
@@ -212,6 +213,8 @@ type ACLResolverConfig struct {
 	Tokens *token.Store
 }
 
+const aclClientDisabledTTL = 30 * time.Second
```
Comment (banks) [inline agent/consul/acl.go:216]: Am I reading this right that the old "default" in `defaults.go` which was actually never overridable was 120 seconds but now we are setting it to be always 30? Does that actually change the behaviour?

If it does, perhaps some commentary in the PR about why the new value is better/safe etc. is worthwhile and a changelog entry would be good? But maybe I'm just misunderstanding the PR as I don't recall this code at all!

Comment (dnephin) [reply agent/consul/acl.go:216]: > Am I reading this right that the old "default" in defaults.go which was actually never overridable was 120 seconds but now we are setting it to be always 30? Does that actually change the behaviour?

No, the 120s was always ignored. See https://github.com/hashicorp/consul/pull/10808#discussion_r684543725 (I've been trying to add that to the PR description but github is having issues).

It was using the 30s set in `NewConsulConfig`, so this preserves the old behaviour.

File: proto/pbconfig/config.proto [hunk 2]
```diff
@@ -43,6 +43,8 @@ message ACL {
    string    DefaultPolicy = 6;
    bool      EnableKeyListPolicy = 7;
    ACLTokens Tokens = 8;
+   // DisabledTTL is deprecated. It is no longer populated and should be ignored
+   // by clients.
    string    DisabledTTL = 9;
```
[the remark this answers is not in the document]
Comment (dnephin) [reply proto/pbconfig/config.proto:48]: Nice! I didn't know about the deprecated field-level option, I'll add that. I also like the idea of renaming it to deprecated, i'll give that a shot.

File: agent/consul/acl_test.go [hunk 3]
```diff
@@ -3580,3 +3580,61 @@ func TestACLResolver_AgentMaster(t *testing.T) {
...
+func TestACLResolver_ACLsEnabled(t *testing.T) {
+	type testCase struct {
+		name     string
+		resolver *ACLResolver
+		enabled  bool
+	}
+
+	run := func(t *testing.T, tc testCase) {
+		require.Equal(t, tc.enabled, tc.resolver.ACLsEnabled())
+	}
```
[the remark this answers is not in the document]
Comment (dnephin) [reply agent/consul/acl_test.go:3593]: Good question! I've been meaning to write up an explanation of this pattern.  I haven't done that yet, so I'll start the write up here. While working on some rather large table tests I  noticed it was difficult to easily navigate between the sections. The common pattern for table tests seems to be something like this:

1. `type testCase`  definition
2. list of test cases
3. for loop
4. test logic

The pattern I've been using (seen in this PR) is this:

1. `type testCase` definition
2. test logic
3. list of test cases
4. for loop

I think this pattern has a key advantage that make it easier to work with large table tests. IDEs make it really easy to jump to the first line of `func TestFoo()`, so generally a test function is going to be read from top to bottom. The list of test cases on its own is difficult to understand, one must first read the "test logic" to see what the test is doing before they can understand the test cases.

By moving the test logic before the test cases the test function can easily be read from top to bottom without having to scroll back and forth through the long list.

The for loop that calls `t.Run` is generally the least interesting part of the test, and is almost always the same boilerplate, so it seems nice to keep that at the very end.

In this particular test function the "test logic" is very small (a single line), so it probably does look a bit strange on first read, but I'm hoping that this pattern makes sense in general, and will start to feel more natural if it becomes more common.

**Review by rboyer — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/429063.

**Pull request merged**